### PR TITLE
Consume secret data from a file.

### PR DESCRIPTION
**Pull request by liranmauda**
### Explain the changes

- Add a util to get data from a file. will use that to fetch Kubernetes secret mounted to a file.
- Replaced env.NOOBAA_AUTH_TOKEN with config.NOOBAA_AUTH_TOKEN
- Replaced env.JWT_SECRET wite config.JWT_SECRET
- Replaced env.SECRET_SERVER with config.SECRET_SERVER
- Keeping the default behavior of using env variables for overriding if needed or testing.

Signed-off-by: liranmauda <liran.mauda@gmail.com>

### Issues: Fixed #xxx / Gap #xxx
1. Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2152143

### Depend on:
- https://github.com/noobaa/noobaa-operator/pull/1015

### CODE REVIEW

File: src/util/os_utils.js [hunk 1]
```diff
@@ -591,16 +592,14 @@ function is_folder_permissions_set(current_path) {
...
-        // for all non kubernetes platforms (docker, local, etc.) return a dummy secret
-        return '12345678';
+    const server_secret = ssl_utils.get_data_from_kubernetes_secret(`/etc/noobaa-server`, `server_secret`);
+    if (server_secret) return server_secret;
+    // in kubernetes we must have SERVER_SECRET loaded from a kubernetes secret
+    if (process.env.CONTAINER_PLATFORM === 'KUBERNETES') {
+        throw new Error('server_secret mounted to a file was not found. it must exist when running in kubernetes');
     }
+    // for all non kubernetes platforms (docker, local, etc.) return a dummy secret
+    return '12345678';
```
Comment (shirady) [inline src/util/os_utils.js:602]: Why do we send a different dummy secret?
`'12345678'` from `read_server_secret`, while `'abcdefgh'` from `get_jwt_secret()`

Comment (liranmauda) [reply src/util/os_utils.js:602]: This is not in the scope of this PR, and I don't want to challenge it here.
This difference was probably created to diverse between server secret and jwt

**Comment by liranmauda** [comment]
Converted to draft due to:
1. We need to get the latest yamls from the operator, so the test environments can work
2. We have one more secretKeyRef to address.

**Comment by liranmauda** [comment]
> Converted to draft due to:
> 
> 1. We need to get the latest yamls from the operator, so the test environments can work
> 2. We have one more secretKeyRef to address.

### Regarding 1:
@dannyzaken should we take the latest yamls from the operator, or should we modify the ones in core? 
Do we have a specific diversity that we would like to keep, or it was just unmaintained? 

### Regarding 2:
It seems that for Postgress we need the env variable so we will keep the secretKeyRef

**Review by dannyzaken — APPROVED** [review]

**Pull request merged**